okio/src/appleMain/kotlin/okio/Source.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+import kotlinx.cinterop.UnsafeNumber
+import kotlinx.cinterop.addressOf
+import kotlinx.cinterop.convert
+import kotlinx.cinterop.reinterpret
+import kotlinx.cinterop.usePinned
+import platform.Foundation.NSInputStream
+import platform.darwin.UInt8Var
+
+/** Returns a source that reads from `in`. */
+fun NSInputStream.source(): Source = NSInputStreamSource(this)
```

Comment:
I think this file should be called `source.kt` (lowercase name) . . . we’ve been doing `lowerCamelFileNames` for files with free functions, `UpperCamelFileNames` when the file matches the primary declaration class.

Reply:
Should `BufferedSource.kt` and `ByteString.kt`, also in this directory, be renamed `bufferedSource.kt`and `byteString.kt` as well? Or perhaps `BufferedSource.kt` should be `inputStream.kt`?

Reply:
@swankjesse after merging the latest upstream changes, the linter doesn't like the lowercase filename `source.kt`, wanting PascalCase. I renamed this file `NSInputStreamSource.kt`. Let me know if there's a preferable name.